### PR TITLE
feat alertmanager: add queryFromGeneratorURL function

### PR DESCRIPTION
**Pull request by FUSAKLA**
Kind of a followup to the previous PR #3849 adding `grafanaExploreURL` function.

To generate the link dynamically, there is no easy way to extract it from the alert. 
The PromQL query is only in the `GeneratorURL` attribute of an alert in the template, and that is URL encoded.
See [related Slack discussion](https://grafana.slack.com/archives/C039863E8P7/p1677158217513759)

So to use it with the previously added  `grafanaExploreURL` this PR adds a new `queryFromGeneratorURL` function which parses is out and decodes it.

Tests, documentation and changelog added

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -57,6 +57,7 @@ Querying with using `{__mimir_storage__="ephemeral"}` selector no longer works.
 * [ENHANCEMENT] Compactor: added the experimental flag `-compactor.block-upload.block-validation-enabled` with the default `true` to configure whether block validation occurs on backfilled blocks. #3411
 * [ENHANCEMENT] Ingester: apply a jitter to the first TSDB head compaction interval configured via `-blocks-storage.tsdb.head-compaction-interval`. Subsequent checks will happen at the configured interval. This should help to spread the TSDB head compaction among different ingesters over the configured interval. #4364
 * [ENHANCEMENT] Ingester: the maximum accepted value for `-blocks-storage.tsdb.head-compaction-interval` has been increased from 5m to 15m. #4364
+* [ENHANCEMENT] Alertmanager: Add additional template function `queryFromGeneratorURL` returning query URL decoded query from the `GeneratorURL` field of an alert. #TODO
```
Comment (FUSAKLA) [inline CHANGELOG.md:60]:
```suggestion
* [ENHANCEMENT] Alertmanager: Add additional template function `queryFromGeneratorURL` returning query URL decoded query from the `GeneratorURL` field of an alert. #4301
```

**Comment by pstibrany** [comment]
The CHANGELOG has just been cut to prepare for the next Mimir release. Please rebase `main` and eventually move the CHANGELOG entry added / updated in this PR to the top of the CHANGELOG document. Thanks!

**Comment by FUSAKLA** [comment]
Hi, thanks for taking a look at this. Sorry, unfortunately I had limited time lately for open source :/

Should be rebased and GTG :crossed_fingers:

**Review by gotjosh — APPROVED** [review]
LGTM

**Comment by k2glyph** [comment]
@mimir-maintainers Can you guys approve this changes?

**Review by pracucci — APPROVED** [review]

**Pull request merged**